test/extensions/filters/listener/tls_inspector/tls_inspector_benchmark.cc

```diff
@@ -0,0 +1,147 @@
+#include <vector>
+
+#include "extensions/filters/listener/tls_inspector/tls_inspector.h"
+
+#include "test/mocks/api/mocks.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/stats/mocks.h"
+#include "test/test_common/threadsafe_singleton_injector.h"
+
+#include "gtest/gtest.h"
+#include "openssl/ssl.h"
+#include "testing/base/public/benchmark.h"
+
+using testing::AtLeast;
+using testing::Invoke;
+using testing::NiceMock;
+using testing::Return;
+using testing::ReturnNew;
+using testing::ReturnRef;
+using testing::SaveArg;
+using testing::_;
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace TlsInspector {
+
+class FastMockListenerFilterCallbacks : public Network::MockListenerFilterCallbacks {
+public:
+  FastMockListenerFilterCallbacks(Network::ConnectionSocket& socket, Event::Dispatcher& dispatcher)
+      : socket_(socket), dispatcher_(dispatcher) {}
+  Network::ConnectionSocket& socket() override { return socket_; }
+  Event::Dispatcher& dispatcher() override { return dispatcher_; }
+  void continueFilterChain(bool success) override { RELEASE_ASSERT(success); }
+
+  Network::ConnectionSocket& socket_;
+  Event::Dispatcher& dispatcher_;
+};
+
+class FastMockConnectionSocket : public Network::MockConnectionSocket {
+public:
+  int fd() const override { return 42; }
+  void setRequestedServerName(absl::string_view server_name) override {
+    if (server_name == "example.com") {
+      got_server_name_ = true;
+    }
+  }
+
+  bool got_server_name_{false};
+};
+
+// Don't inherit from the mock implementation at all, because this is instantiated
+// in the hot loop.
+class FastMockFileEvent : public Event::FileEvent {
+  void activate(uint32_t) override {}
+  void setEnabled(uint32_t) override {}
+};
+
+class FastMockDispatcher : public Event::MockDispatcher {
+public:
+  Event::FileEventPtr createFileEvent(int, Event::FileReadyCb cb, Event::FileTriggerType,
+                                      uint32_t) override {
+    file_event_callback_ = cb;
+    return std::make_unique<FastMockFileEvent>();
+  }
+
+  Event::FileReadyCb file_event_callback_;
+};
+
+class FastMockOsSysCalls : public Api::MockOsSysCalls {
+public:
+  FastMockOsSysCalls(const std::vector<uint8_t>& client_hello) : client_hello_(client_hello) {}
+
+  ssize_t recv(int, void* buffer, size_t length, int) override {
+    RELEASE_ASSERT(length >= client_hello_.size());
+    memcpy(buffer, client_hello_.data(), client_hello_.size());
+    return client_hello_.size();
+  }
+
+  const std::vector<uint8_t> client_hello_;
+};
+
+static std::vector<uint8_t> generateClientHello(const std::string& sni_name) {
```

Comment:
Is this the same in both tests?  Seems like a good general test utilitiy to me.

Reply:
Should I put that in test_common/utility, or create a separate library for it?

Reply:
Separate library, please.    I was about to suggest it go in test/extensions/filters/listener/tls_inspector/ but of course this is clearly useful for other tls code.  As long as we have it split out into its own library/files we could theoretically build-rule it out.